src/Tickets/Commerce/Cart/Unmanaged_Cart.php

```diff
@@ -259,9 +276,18 @@ public function process( array $data = [] ) {
 	}
 
 	/**
-	 * {@inheritDoc}
+	 * Insert or update an item.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|int $item_id     The item ID.
+	 * @param int        $quantity    The quantity of the item. If the item exists, this quantity will override
+	 *                                the previous quantity. Passing 0 will remove the item from the cart entirely.
+	 * @param array      $extra_data  Extra data to save to the item.
+	 *
+	 * @return void
 	 */
-	public function prepare_data( array $data = [] ) {
-		return $data;
+	public function upsert_item( $item_id, int $quantity, array $extra_data = [] ) {
```

Comment:
This is now a useless function.

Reply:
Yes, it's maintaining the way this class has functioned with minimal changes. All the real changes are in the `Agnostic_Cart` class.